### PR TITLE
fix: scroll was changed in scg

### PR DESCRIPTION
**Pull request by andrewKapustin22**
Added a new feature, now when zooming in scg, the increase occurs at the cursor

### CODE REVIEW

**Comment by ChaikaS** [comment]
update change log

File: components/scg/src/scg-render.js [hunk 1]
```diff
@@ -14,6 +14,23 @@ SCg.Render.prototype = {
...
+            const svg = e.currentTarget
+            const svgRect = svg.getBoundingClientRect();
+
+            const svgX = e.clientX - svgRect.x;
+            const svgY = e.clientY - svgRect.y;
+
+            const transformX = (svgX - this.translate[0]) / this.scale;
+            const transformY = (svgY - this.translate[1]) / this.scale;
+
+            e.wheelDelta > 0 ? this.scale *= 1.1 : this.scale *= 0.9;
```
Comment (ChaikaS) [inline components/scg/src/scg-render.js:27]: use variables for these numbers

File: components/scg/src/scg-render.js [hunk 2]
```diff
@@ -40,10 +59,14 @@ SCg.Render.prototype = {
                 self.scene.onMouseUpObject(d);
                 if (d3.event.stopPropagation()) d3.event.stopPropagation();
             })
-            .on("wheel", function(){
+            .on("wheel", function () {
```
Comment (ChaikaS) [inline components/scg/src/scg-render.js:62]: here and beyond, check linter please

**Review by ChaikaS — APPROVED** [review]

**Comment by ChaikaS** [comment]
work well

File: components/scg/src/scg-render.js [hunk 3]
```diff
@@ -13,7 +13,26 @@ SCg.Render.prototype = {
...
+
+            const svgX = e.clientX - svgRect.x;
+            const svgY = e.clientY - svgRect.y;
+
+            const transformX = (svgX - this.translate[0]) / this.scale;
+            const transformY = (svgY - this.translate[1]) / this.scale;
+
+            e.wheelDelta > 0 ? this.scale *= this.zoomIn : this.scale *= this.zoomOut;
+
+            this.translate[0] = svgX - transformX * this.scale;
```
Comment (smousy) [inline components/scg/src/scg-render.js:31]: indexes 0 and 1 strange, maybe some meaningful name?

**Review by smousy — APPROVED** [review]

File: components/scg/src/scg-render.js (hunk 3 above)
Comment (smousy) [reply components/scg/src/scg-render.js:31]: use variable

File: components/scg/src/scg-render.js [hunk 4]
```diff
@@ -39,10 +58,14 @@ SCg.Render.prototype = {
...
                 if (d3.event.stopPropagation()) d3.event.stopPropagation();
             })
-            .on("wheel", function(){
+            .on("wheel", function () {
                 var direction = d3.event.wheelDelta < 0 ? 'down' : 'up';
-                if (direction === 'up') self.scene.onMouseWheelUp(this, self);
-                if (direction === 'down') self.scene.onMouseWheelDown(this, self);
+                if (direction === 'up') {
+                    self.transformByZoom(d3.event);
+                };
```
Comment (MikhailSadovsky) [inline components/scg/src/scg-render.js:65]:
```suggestion
                }
```
Please fix here and later

**Review by MikhailSadovsky — APPROVED** [review]

**Pull request merged**